service/storage_service.cc

```diff
@@ -2944,9 +2944,9 @@ future<> storage_service::restore_replica_count(inet_address endpoint, inet_addr
     });
     auto streamer = make_lw_shared<dht::range_streamer>(_db, _stream_manager, tmptr, as, get_broadcast_address(), _sys_ks.local().local_dc_rack(), "Restore_replica_count", streaming::stream_reason::removenode);
     removenode_add_ranges(streamer, endpoint).get();
-    auto status_checker = seastar::async([this, endpoint, &as] {
-        slogger.info("restore_replica_count: Started status checker for removing node {}", endpoint);
-        while (!as.abort_requested()) {
+    slogger.info("restore_replica_count: Starting status checker for removing node {}", endpoint);
+    auto status_checker = do_until([&as] { return as.abort_requested(); }, [this, endpoint, &as] {
+            // FIXME: indentation
```

Comment:
Why not a coroutine?

Reply:
We can, but isn't it an overkill?

Reply:
Maybe it might be worth it, to reduce the churn...